src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -263,4 +293,131 @@ describe("WalletOverviewPage", () => {
+    spyOn(comp, "getWalletBalance");
+    spyOn(comp, "getTransactionHistory");
+
+    comp.currentWallet = <IWallet>{};
+    comp.currentWallet.publicKey = "";
+
+    comp.refreshWalletBalance();
+
+    expect(comp.getWalletBalance).toHaveBeenCalled();
+    expect(comp.getTransactionHistory).toHaveBeenCalled();
+  })
+
+  it("should not open landing page and show a toast after deleting a wallet while some wallets still exist", (done) => {
+    // Get all 3 wallets
+    comp.getAllWallets().then(data => {
+      // Delete first one
+      comp.deleteSelectedWallet(comp.currentWallet);
+
+      spyOn(comp, "openLandingPage");
+      spyOn(comp, "showToastMessage");
+
+      // Delete last one
+      comp.deleteSelectedWallet(comp.currentWallet);
+
+      expect(comp.openLandingPage).not.toHaveBeenCalled();
+      expect(comp.showToastMessage).not.toHaveBeenCalled();
+      
+      done();
+    });
+  })
+
+  it("should return 7 fixed numbers if the picked currency is not in the if condition", () => {
+    comp.pickedCurrency = "I DON'T EXIST";
+
+    let result = comp.getFixedNumbers();
+
+    expect(result).toEqual(7);
+  })
+
+  it("should call createElement, appendChild, execCommand and removechild upon copying to clipboard for web", () => {
+    spyOn(document, "createElement");
+    spyOn(document.body, "appendChild");
+    spyOn(document, "execCommand");
+    spyOn(document.body, "removeChild");
+
+    comp.copyToClipboardWeb("data");
+
+    expect(document.createElement).toHaveBeenCalled();
+    expect(document.body.appendChild).toHaveBeenCalled();
+    expect(document.execCommand).toHaveBeenCalled();
+    expect(document.body.removeChild).toHaveBeenCalled();
+  })
+
+  it("should call createElement, appendChild, removeChild upon downloading a txt file for web", () => {
+    spyOn(document, "createElement");
+    spyOn(document.body, "appendChild");
+    spyOn(document.body, "removeChild");
+
+    comp.downloadTxtFileWeb("data", "filename");
+
+    expect(document.createElement).toHaveBeenCalled();
+    expect(document.body.appendChild).toHaveBeenCalled();
+    expect(document.body.removeChild).toHaveBeenCalled();
+  });
+
+  it("should call create and present upon showing a toast message", () => {
+    spyOn(toastController, "create").and.returnValue(toast);
+    spyOn(toast, "present");
+    
+    comp.showToastMessage("message", 1000, "bottom");
+
+    expect(toastController.create).toHaveBeenCalled();
+    expect(toast.present).toHaveBeenCalled();
+  });
+
+  it("should call a create and present error modal after getting the available exchanges list with a rejected promise", (done) => {
+    spyOn(exchangeService, "getAvailableExchanges").and.returnValue(Promise.reject(""));
+    spyOn(comp, "getAvailableExchanges").and.callThrough();
+    spyOn(alertController, "create").and.returnValue(alert);
+    spyOn(alert, "present");
+
+    comp.initialize().then(data => {
```

Comment:
Why not call the `getAvailableExchanges` method on the controller? Initialize does not purely retrieve the exchanges right?